src/app/components/button/button.ts

```diff
@@ -205,13 +205,10 @@ export class ButtonDirective implements AfterViewInit, OnDestroy {
         let iconElement = DomHandler.findSingle(this.htmlElement, '.p-button-icon');
         let labelElement = DomHandler.findSingle(this.htmlElement, '.p-button-label');
 
-        if (!this.icon && !this.loading) {
-            iconElement && this.htmlElement.removeChild(iconElement);
-            return;
-        }
-
         if (this.loading && !this.loadingIcon && iconElement) {
             iconElement.innerHTML = this.spinnerIcon;
+        } else if (iconElement?.innerHTML) {
```

Comment:
Oh for sure, I have zero preference. Just would really like to see this make it in the next release since it's blocking us from upgrading since all of our buttons are use the pButton directive instead of the p-button element.

Though I think to avoid just undoing the loading spinner it would have to be:
```
if (this.loading && !this.loadingIcon && iconElement) {
    iconElement.innerHTML = this.spinnerIcon;
}

if (!this.loading && iconElement?.innerHTML) {
    iconElement.innerHTML = '';
}
```

Reply:
I did a PR fixing something similar in `pButton` and `<p-button` time ago https://github.com/primefaces/primeng/pull/13363

Maybe I forgot something and with your code I hope works 100%